core/field_angle.js

```diff
@@ -135,7 +135,9 @@ Blockly.FieldAngle.prototype.DEFAULT_VALUE = 0;
  * @nocollapse
  */
 Blockly.FieldAngle.fromJson = function(options) {
-  return new Blockly.FieldAngle(options['angle'], undefined, options);
+  // `this` might be a subclass of FieldAngle if that class doesn't override
+  // the static fromJson method.
+  return new this(options['angle'], undefined, options);
```

Comment:
I'm a little concerned about this. There's nothing guaranteeing that a subclass constructor has the same method signature right? Eg say someone wants to define a FieldAngle subclass that has a default angle of 90 degrees:
```
// No value parameter to constructor
Blockly.FieldAngleNinety = function(opt_validator, opt_config) { ... }
```
And they also forget to define a fromJson method. It might be tricky for them to debug what's going on. But I'm not sure of a better way to handle this either... thoughts?

Reply:
At the moment if FieldAngleNinety forgets to define a fromJson, in ES6 a FieldAngle will be unexpectedly created instead.  This is hilariously challenging to debug (took me nearly two days to figure out what was going on).  So this PR will at least create the right object.

I'm less concerned about a change in signature.  After all, what we are describing is someone creating a new field, inventing a new JSON signature out of thin air, but forgetting to actually code that signature anywhere.  It's a bit like calling Blockly.inject with some new options, but not actually adding those options to the codebase.

Reply:
Good point. LGTM then :D